platforms/ios/framework/src/TGCameraPosition.mm

```diff
@@ -0,0 +1,90 @@
+//
+//  TGCameraPosition.m
+//  TangramMap
+//
+//  Created by Matt Blair on 7/18/18.
+//
+
+#import "TGCameraPosition.h"
+#import "TGCameraPosition+Internal.h"
+#import <Foundation/Foundation.h>
+
+#include "map.h"
+
+@implementation TGCameraPosition
+
+- (instancetype)initWithCenter:(CLLocationCoordinate2D)center
+                          zoom:(CGFloat)zoom
+                       bearing:(CLLocationDirection)bearing
+                         pitch:(CGFloat)pitch
+{
+    self = [super init];
+    if (self) {
+        _center = center;
+        _zoom = zoom;
+        _bearing = bearing;
+        _pitch = pitch;
+    }
+    return self;
+}
+
+- (instancetype)initWithCoreCamera:(Tangram::CameraPosition *)camera
+{
+    self = [super init];
+    if (self) {
+        [self setToCoreCamera:camera];
+    }
+    return self;
+}
+
+- (void)setToCoreCamera:(Tangram::CameraPosition *)camera
+{
+    if (camera == nullptr) {
+        return;
+    }
+    _center.longitude = camera->longitude;
+    _center.latitude = camera->latitude;
+    _zoom = camera->zoom;
+    _bearing = (-180.0 / M_PI) * camera->rotation;
+    _pitch = (180.0 / M_PI) * camera->tilt;
+}
+
+- (Tangram::CameraPosition)convertToCoreCamera
+{
+    Tangram::CameraPosition camera;
+    camera.longitude = _center.longitude;
+    camera.latitude = _center.latitude;
+    camera.zoom = _zoom;
+    camera.rotation = (-M_PI / 180.0) * _bearing;
+    camera.tilt = (M_PI / 180.0) * _pitch;
+    return camera;
+}
+
+- (id)copyWithZone:(NSZone *)zone
```

Comment:
I do not know about `NSZone`, but based on the document (https://developer.apple.com/documentation/foundation/nszone?language=objc), its not recommended to be used.

Reply:
It's just what has to be implemented to satisfy the `NSCopying` protocol, which is typical of data classes (https://developer.apple.com/documentation/foundation/nscopying?language=objc)